tests/e2e/sentinel/utils.js

```diff
@@ -75,6 +123,21 @@ const getInfoDocs = (docIds = []) => {
   return requestOnSentinelTestDb(opts).then(response => response.rows.map(row => row.doc));
 };
 
+//for e2e tests
+const getInfoDocsNative = async (docIds = []) => {
+  docIds = Array.isArray(docIds) ? docIds : [docIds];
+
+  const opts = {
+    path: '/_all_docs?include_docs=true',
+    body: { keys: docIds.map(id => id + '-info') },
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+  };
+  return await requestOnSentinelTestDbNative(opts).then(response => response.rows.map(row => row.doc));
```

Comment:
I think this should be a little different because there is a mix of `.then` and await.  This will wait for the request promise to complete and then you return the new array from `.map`

```
 const response = await requestOnSentinelTestDbNative(opts)
 return response.rows.map(row => row.doc));
 ```